### PR TITLE
feat(build): support for multi arch container image

### PR DESCRIPTION
**Pull request by prateekpandey14**
Signed-off-by: prateekpandey14 <prateek.pandey@mayadata.io>

**Why is this PR required? What issue does it fix?**:
PR add multi arch support for zfs-localpv container images

**Does this PR require any upgrade changes?**:

**If the changes in this PR are manually verified, list down the scenarios covered:**:

**Any additional information for your reviewer?** :
_Mention if this PR is part of any design or a continuation of previous PRs_

**Checklist:**
- [ ] Fixes #<issue number>
- [ ] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Has the change log section been updated?
- [ ] Commit has unit tests
- [ ] Commit has integration tests
- [ ] (Optional) Are upgrade changes included in this PR? If not, mention the issue/PR to track:
- [ ] (Optional) If documentation changes are required, which issue on https://github.com/openebs/openebs-docs is used to track them:

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        run: |
+          BRANCH="${GITHUB_REF##*/}"
+          CI_TAG=${BRANCH#v}-ci
+          if [ ${BRANCH} = "master" ]; then
+            CI_TAG="ci"
+          fi
+          echo "::set-env name=TAG::${CI_TAG}"
+          echo "::set-env name=BRANCH::${BRANCH}"
+          echo "BRANCH: ${BRANCH}"
+          echo "TAG: ${CI_TAG}"
```
Comment (akhilerm) [inline .github/workflows/build.yml:64]:
```suggestion
          echo "TAG: ${CI_TAG}"
          
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+
+      - name: Set Tag
+        run: |
+          TAG="${GITHUB_REF#refs/*/v}"
+          echo "::set-env name=TAG::${TAG}"
+          echo "::set-env name=RELEASE_TAG::${TAG}"
+          echo "RELEASE_TAG ${TAG}"
```
Comment (akhilerm) [inline .github/workflows/release.yml:33]:
```suggestion
          echo "RELEASE_TAG ${TAG}"
          
```

File: Makefile.buildx.mk [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+# Experimental docker feature to build cross platform multi-architecture docker images
+# https://docs.docker.com/buildx/working-with-buildx/
+
+ifeq (${TAG}, )
+  export TAG=ci
+endif
+
+# default list of platforms for which multiarch image is built
+ifeq (${PLATFORMS}, )
+	export PLATFORMS="linux/amd64,linux/arm64,linux/ppc64le"
```
Comment (akhilerm) [inline Makefile.buildx.mk:26]: why is armv7 not included?

**Review by akhilerm — COMMENTED** [review]
a few nitpick comments and a question on the `linux/arm/v7` platform.

File: Makefile.buildx.mk (hunk 3 above)
Comment (prateekpandey14) [reply Makefile.buildx.mk]: added

**Review by shubham14bajpai — APPROVED** [review]
lgtm

**Review by akhilerm — APPROVED** [review]
/lgtm

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        id: buildx
+        uses: docker/setup-buildx-action@v1
+        with:
+          version: latest
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
```
Comment (akhilerm) [inline .github/workflows/build.yml:81]:
```suggestion
          username: ${{ secrets.DOCKERHUB_USERNAME }}
          password: ${{ secrets.DOCKERHUB_TOKEN }}
```

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        id: buildx
+        uses: docker/setup-buildx-action@v1
+        with:
+          version: latest
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v1 
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
```
Comment (akhilerm) [inline .github/workflows/release.yml:50]:
```suggestion
          username: ${{ secrets.DOCKERHUB_USERNAME }}
          password: ${{ secrets.DOCKERHUB_TOKEN }}
```

**Review by akhilerm — CHANGES_REQUESTED** [review]
@prateekpandey14 A change is required for the dockerhub credential env to make it consistent across all repositories.

File: .github/workflows/release.yml (hunk 5 above)
Comment (prateekpandey14) [reply .github/workflows/release.yml]: done

**Review by akhilerm — APPROVED** [review]
LGTM

**Comment by kmova** [comment]
@prateekpandey14 - can you get travis to push the zfs-driver-amd64 image?

**Comment by prateekpandey14** [comment]
>@prateekpandey14 - can you get travis to push the zfs-driver-amd64 image?

Done

**Review by kmova — APPROVED** [review]

**Comment by pawanpraka1** [comment]
@prateekpandey14 can you raise a PR for the changelog?

**Pull request merged**